apps/meteor/server/services/team/service.ts

```diff
@@ -472,12 +483,16 @@ export class TeamService extends ServiceClassInternal implements ITeamService {
 		room.teamDefault = isDefault;
 		await Rooms.setTeamDefaultById(rid, isDefault);
 
-		if (room.teamDefault) {
-			const teamMembers = await this.members(uid, room.teamId, true, undefined, undefined);
+		if (isDefault) {
+			const teamMembers = await this.members(uid, room.teamId, true, { offset: 0, count: MAX_NUMBER_OF_AUTO_JOIN_MEMBERS });
 
 			for await (const m of teamMembers.records) {
 				await addUserToRoom(room._id, m.user, user);
 			}
+			return {
+				room,
+				numberOfMembers: teamMembers.total,
+			};
```

Comment:
```suggestion
```
I don't think we need this new `numberOfMembers` field to be returned here. You can use `room.usersCount` instead, we already store this info in the room.  By doing that, you also won't need multiple return types for this method.

Reply:
You're right, the approach I took also has a subtle problem I just found out, that the channel could have more users than the number of members in the team, so the number would be messed up.